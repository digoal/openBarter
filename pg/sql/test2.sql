SET client_min_messages = warning;
\set ECHO none
-- CREATE LANGUAGE PLPGSQL;
\i openbarter.sql
\set ECHO all
RESET client_min_messages;
/*
truncate table tlomega restart identity cascade;
truncate table tomega restart identity cascade;

truncate table tcommit restart identity ;
truncate table tdraft restart identity cascade;

truncate table tnoeud restart identity cascade;
truncate table tstock restart identity cascade;
truncate table towner restart identity cascade;
truncate table tquality restart identity cascade;
-- truncate table tdepositary restart identity cascade;
truncate table tmvt restart identity cascade;
truncate table tldraft restart identity cascade;
select setval('tdraft_id_seq',1);
*/
-- insert into towner (name) values ('market');

select fcreate_quality('q1');
select fcreate_quality('q2');
select fcreate_quality('q3');
select fcreate_quality('q4');
select fcreate_quality('q5');
select fcreate_quality('q6');

-- owners are created
SELECT fadd_account('o1','olivier>q1',1000); 
SELECT fadd_account('o1','olivier>q2',1000); 
SELECT fadd_account('o1','olivier>q3',1000); 
SELECT fsub_account('o1','olivier>q1',1000); 
SELECT fsub_account('o1','olivier>q2',1000); 
SELECT fsub_account('o1','olivier>q3',500); 

-- should be 0
SELECT corrupted_stock_a+corrupted_stock_s+unbananced_qualities+corrupted_draft as errors from fstats();

select qown,qname,owner,qtt from vowned;
select qown,qname,qtt from vbalance;
select id,did,provider,nat,qtt,receiver from vmvt;




