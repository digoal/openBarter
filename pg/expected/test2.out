SET client_min_messages = warning;
\set ECHO none
 setval 
--------
      1
(1 row)

RESET client_min_messages;
/*
truncate table tlomega restart identity cascade;
truncate table tomega restart identity cascade;

truncate table tcommit restart identity ;
truncate table tdraft restart identity cascade;

truncate table tnoeud restart identity cascade;
truncate table tstock restart identity cascade;
truncate table towner restart identity cascade;
truncate table tquality restart identity cascade;
-- truncate table tdepositary restart identity cascade;
truncate table tmvt restart identity cascade;
truncate table tldraft restart identity cascade;
select setval('tdraft_id_seq',1);
*/
-- insert into towner (name) values ('market');
select fcreate_quality('q1');
 fcreate_quality 
-----------------
               0
(1 row)

select fcreate_quality('q2');
 fcreate_quality 
-----------------
               0
(1 row)

select fcreate_quality('q3');
 fcreate_quality 
-----------------
               0
(1 row)

select fcreate_quality('q4');
 fcreate_quality 
-----------------
               0
(1 row)

select fcreate_quality('q5');
 fcreate_quality 
-----------------
               0
(1 row)

select fcreate_quality('q6');
 fcreate_quality 
-----------------
               0
(1 row)

-- owners are created
SELECT fadd_account('o1','olivier>q1',1000); 
 fadd_account 
--------------
            0
(1 row)

SELECT fadd_account('o1','olivier>q2',1000); 
 fadd_account 
--------------
            0
(1 row)

SELECT fadd_account('o1','olivier>q3',1000); 
 fadd_account 
--------------
            0
(1 row)

SELECT fsub_account('o1','olivier>q1',1000); 
 fsub_account 
--------------
            0
(1 row)

SELECT fsub_account('o1','olivier>q2',1000); 
 fsub_account 
--------------
            0
(1 row)

SELECT fsub_account('o1','olivier>q3',500); 
 fsub_account 
--------------
            0
(1 row)

-- should be 0
SELECT corrupted_stock_a+corrupted_stock_s+unbananced_qualities+corrupted_draft as errors from fstats();
 errors 
--------
      0
(1 row)

select qown,qname,owner,qtt from vowned;
  qown   |   qname    | owner | qtt 
---------+------------+-------+-----
 olivier | olivier>q3 | o1    | 500
(1 row)

select qown,qname,qtt from vbalance;
  qown   |   qname    | qtt 
---------+------------+-----
 olivier | olivier>q3 | 500
(1 row)

select id,did,provider,nat,qtt,receiver from vmvt;
 id | did | provider |    nat     | qtt  | receiver 
----+-----+----------+------------+------+----------
  1 |     | market   | olivier>q1 | 1000 | o1
  2 |     | market   | olivier>q2 | 1000 | o1
  3 |     | market   | olivier>q3 | 1000 | o1
  4 |     | o1       | olivier>q1 | 1000 | market
  5 |     | o1       | olivier>q2 | 1000 | market
  6 |     | o1       | olivier>q3 |  500 | market
(6 rows)

